src/Controllers/MoviesController.cs

```diff
@@ -45,9 +45,12 @@ public IActionResult Create(MovieViewModel mvm)
         }
 
         [HttpGet("Index")]
-        public IActionResult Index(string movieGenre, string searchString) 
+        public IActionResult Index(string movieGenre, string searchString, string sortOrder) 
         {
             var movies = UnitOfWork.Movies.GetAll().ToList();
+            ViewData["TitleSortParam"] = String.IsNullOrEmpty(sortOrder) ? "title_desc" : "";
+            ViewData["ReleaseDateSortParam"] = sortOrder == "Release Date" ? "date_desc" : "ReleaseDate";
+            ViewData["GenreSortParam"] = sortOrder == "Genre" ? "genre_desc" : "Genre";
```

Comment:
Regarding line 54: It is not efficient to make a query of all the movies (line 50) and then extract the genres to use them in another query in the table movies (again). Try to use Linq :)